### ISSUE TITLE
Vulkan and Wayland support.

### ISSUE THREAD

**Issue by DmitryHetman** [vurtun/nuklear#411]
Add Wayland support and Vulkan API for rendering.

**Comment by DmitryHetman** [comment]
[Sellers G. - Vulkan Programming Guide The Official Guide to Learning Vulkan - 2017.pdf](https://github.com/vurtun/nuklear/files/920284/Sellers.G.-.Vulkan.Programming.Guide.The.Official.Guide.to.Learning.Vulkan.-.2017.pdf)

**Comment by DmitryHetman** [comment]
https://jan.newmarch.name/Wayland/index.html

**Comment by vurtun** [comment]
No, I will not add more backends. Feel free to create a PR but I have no interest at the moment to learn two additional frameworks with one being one of the most complex APIs I have ever seen.

**Comment by m0ppers** [comment]
FYI I have started doing a vulkan renderer in case that is still relevant: https://github.com/m0ppers/nuklear-vulkan

**Comment by snuk182** [comment]
Yes please @m0ppers ! Huge thanks in advance.

**Comment by skoperst** [comment]
I added initial Wayland support in a pull-request. That's about half way for vulkan-wayland

https://github.com/vurtun/nuklear/pull/926
